test/e2e/windows/memory_limits.go

```diff
@@ -104,32 +104,38 @@ func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int)
 	const (
 		podType = "memory_limit_test_pod"
 	)
-
-	totalAllocatable := getTotalAllocatableMemory(f)
-
-	memValue := totalAllocatable.Value()
-	memPerPod := memValue / int64(allocatablePods)
-	ginkgo.By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
-
-	// these should all work
-	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().CreateBatch(pods)
-
-	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().Create(failurePods[0])
-
-	gomega.Eventually(func() bool {
-		eventList, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
-		framework.ExpectNoError(err)
-		for _, e := range eventList.Items {
-			// Look for an event that shows FailedScheduling
-			if e.Type == "Warning" && e.Reason == "FailedScheduling" && e.InvolvedObject.Name == failurePods[0].ObjectMeta.Name {
-				framework.Logf("Found %+v event with message %+v", e.Reason, e.Message)
-				return true
+	//
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
+	totalAllocatable := resource.NewQuantity(0, resource.BinarySI)
+	for _, node := range nodeList.Items {
+		status := node.Status
+		totalAllocatable.Add(status.Allocatable[v1.ResourceMemory])
+		nodeAllocatable := totalAllocatable
+
+		pods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(nodeAllocatable, 1))
+		f.PodClient().CreateBatch(pods[0])
+
+		failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(nodeAllocatable, 1))
+		f.PodClient().CreateBatch(failurePods[0])
+
+		gomega.Eventually(func() bool {
```

Comment:
this should be outside this loop. It is used to verify that a pod will not be scheduled if there is no allocatable memory available left for nodes